### PR TITLE
Unique ids

### LINKED ISSUE #52: Unique identifiers when saving
Every process and every file should get a unique identifier in the generated records.

For processes, since their `pid` is unique, I have been using:
```C
sprintf(pbuf, "pid%ld", (long) pid);
```

But what about files?

### PR DESCRIPTION
**Pull request by fvalasiad**
Closes #52

### CODE REVIEW

File: src/types.h [hunk 1]
```diff
@@ -24,9 +24,10 @@ SPDX-License-Identifier: LGPL-2.1-or-later
  * Not all info is added at the same time.
  */
 typedef struct {
+    char outname[33];		  // Up to 10^32 - 1
```
Comment (zvr) [inline src/types.h:27]: make it 16, if it's an array.
I'd make it a `char *` and assign the result of a `strdup`, but a `char []` is still fine.

File: src/types.h [hunk 2]
```diff
@@ -38,6 +39,7 @@ typedef struct {
  * stop info struct.
  */
 typedef struct {
+    char outname[33];
```
Comment (zvr) [inline src/types.h:42]: this should also be 16

**Review by zvr — CHANGES_REQUESTED** [review]

File: src/types.h (hunk 2 above)
Comment (fvalasiad) [reply src/types.h:42]: @zvr isn't that too low? `ulimit -Hn` returns 524288 in my system. I am not against dynamic memory but this is easily solve-able with a few extra bytes of static memory.

Comment (zvr) [reply src/types.h:42]: Even the whole Linux kernel has less than 100K files; even a `[8]` would be fine.
A `[16]` will probably cover a build of a whole distribution.

Comment (fvalasiad) [reply src/types.h:42]: @zvr Oh okay cool to know some statistics. I will make it 17 tho account for the 1 byte reserved for `[fp]` :)

File: src/types.h [hunk 3]
```diff
@@ -24,9 +24,10 @@ SPDX-License-Identifier: LGPL-2.1-or-later
  * Not all info is added at the same time.
  */
 typedef struct {
+    char outname[17];
```
Comment (zvr) [inline src/types.h:27]: 16, please -- that was the whole point

File: src/types.h [hunk 4]
```diff
@@ -38,6 +39,7 @@ typedef struct {
  * stop info struct.
  */
 typedef struct {
+    char outname[17];
```
Comment (zvr) [inline src/types.h:42]: 16 here as well

**Review by zvr — CHANGES_REQUESTED** [review]

**Review by zvr — APPROVED** [review]

**Pull request merged**